### DISCUSSION TITLE
Problem in the browser console

### DISCUSSION THREAD

**Discussion by FanYueee** [thuliteio/doks#994] [Q&A] [answered]
I found the following two problems while checking and I deployed the site in Cloudflare Pages.
This seems to be related to the file uUFfzeenU-uE98DMGiMkO3KxqtU.js and https://*.netlify.app

Refused to load the stylesheet 'data:text/css;charset=utf-8;base64,LmVjdS13cmFwcGVyCnsKCXotaW5kZXg6IDUwMDAwOwoJcG9zaXRpb246IGZpeGVkOwoJd2lkdGg6IDUwcHg7CgloZWlnaHQ6IDUwcHg7CgliYWNrZ3JvdW5kOiBub25lOwoJcGFkZGluZzogMDsKfQouZWN1LXdyYXBwZXIsIC5lY3Utd3JhcHBlciAqCnsKCWxpbmUtaGVpZ2h0OiBub3JtYWw7CgltYXJnaW46IDA7Cn0KCi5lY3Utd3JhcHBlci50b3AtbGVmdAp7CglsZWZ0OiAxNXB4OwoJdG9wOiAxNXB4Owp9Ci5lY3Utd3JhcHBlci50b3AtcmlnaHQKewoJdG9wIDogMTVweDsKCXJpZ2h0OiAxNXB4Owp9Ci5lY3Utd3JhcHBlci5ib3R0b20tbGVmdAp7Cglib3R0b206IDE1cHg7CglsZWZ0OiAxNXB4Owp9Ci5lY3Utd3JhcHBlci5i...BweDsKCW1heC1oZWlnaHQ6IDQwcHg7CglwYWRkaW5nOiA1cHggMTBweDsKCXdoaXRlLXNwYWNlOiBub3dyYXA7Cgl0ZXh0LW92ZXJmbG93OiBlbGxpcHNpczsKCW92ZXJmbG93OiBoaWRkZW47CgltYXJnaW4tdG9wOiAxMHB4OwoJdHJhbnNpdGlvbjogb3BhY2l0eSAzMDBtcyBsaW5lYXI7CglvcGFjaXR5OiAwOwoJdGV4dC1hbGlnbjogY2VudGVyOwp9Ci5tZXNzYWdlLnNob3cKewoJb3BhY2l0eTogMTsKfQoubWVzc2FnZS50b3AtbGVmdAp7CglsZWZ0OiA2MHB4Owp9Ci5tZXNzYWdlLnRvcC1yaWdodAp7CglyaWdodDogNjBweDsKfQoubWVzc2FnZS5ib3R0b20tbGVmdAp7CglsZWZ0OiA2MHB4Owp9Ci5tZXNzYWdlLmJvdHRvbS1yaWdodAp7CglyaWdodDogNjBweDsKfQ==' because it violates the following Content Security Policy directive: "style-src 'self' https://*.netlify.app 'unsafe-inline'". Note that 'style-src-elem' was not explicitly set, so 'style-src' is used as a fallback.

Refused to load the stylesheet 'data:text/css;charset=utf-8;base64,LmVjdS13cmFwcGVyCnsKCXotaW5kZXg6IDUwMDAwOwoJcG9zaXRpb246IGZpeGVkOwoJd2lkdGg6IDUwcHg7CgloZWlnaHQ6IDUwcHg7CgliYWNrZ3JvdW5kOiBub25lOwoJcGFkZGluZzogMDsKfQouZWN1LXdyYXBwZXIsIC5lY3Utd3JhcHBlciAqCnsKCWxpbmUtaGVpZ2h0OiBub3JtYWw7CgltYXJnaW46IDA7Cn0KCi5lY3Utd3JhcHBlci50b3AtbGVmdAp7CglsZWZ0OiAxNXB4OwoJdG9wOiAxNXB4Owp9Ci5lY3Utd3JhcHBlci50b3AtcmlnaHQKewoJdG9wIDogMTVweDsKCXJpZ2h0OiAxNXB4Owp9Ci5lY3Utd3JhcHBlci5ib3R0b20tbGVmdAp7Cglib3R0b206IDE1cHg7CglsZWZ0OiAxNXB4Owp9Ci5lY3Utd3JhcHBlci5i...BweDsKCW1heC1oZWlnaHQ6IDQwcHg7CglwYWRkaW5nOiA1cHggMTBweDsKCXdoaXRlLXNwYWNlOiBub3dyYXA7Cgl0ZXh0LW92ZXJmbG93OiBlbGxpcHNpczsKCW92ZXJmbG93OiBoaWRkZW47CgltYXJnaW4tdG9wOiAxMHB4OwoJdHJhbnNpdGlvbjogb3BhY2l0eSAzMDBtcyBsaW5lYXI7CglvcGFjaXR5OiAwOwoJdGV4dC1hbGlnbjogY2VudGVyOwp9Ci5tZXNzYWdlLnNob3cKewoJb3BhY2l0eTogMTsKfQoubWVzc2FnZS50b3AtbGVmdAp7CglsZWZ0OiA2MHB4Owp9Ci5tZXNzYWdlLnRvcC1yaWdodAp7CglyaWdodDogNjBweDsKfQoubWVzc2FnZS5ib3R0b20tbGVmdAp7CglsZWZ0OiA2MHB4Owp9Ci5tZXNzYWdlLmJvdHRvbS1yaWdodAp7CglyaWdodDogNjBweDsKfQ==' because it violates the following Content Security Policy directive: "style-src 'self' https://*.netlify.app 'unsafe-inline'". Note that 'style-src-elem' was not explicitly set, so 'style-src' is used as a fallback.

**Comment by h-enk** [comment, marked as answer]
> Refused to load the stylesheet 'data::[..]' because it violates the following Content Security Policy directive: "style-src 'self' https://*.netlify.app 'unsafe-inline'" [..].


So, add `data:` to `style-src` in `./layouts/index.headers/` like so:

```bash
/*
  Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
  X-Content-Type-Options: nosniff
  X-XSS-Protection: 1; mode=block
  Content-Security-Policy: default-src 'self'; frame-ancestors https://jamstackthemes.dev; manifest-src 'self' https://*.netlify.app; connect-src 'self' https://*.netlify.app; font-src 'self' https://*.netlify.app; img-src 'self' https://*.netlify.app data: https://i.giphy.com; script-src 'self' https://*.netlify.app 'sha512-RGGByJUOP98hE4wFZM78RM/3MijWJs0Tm0DbfrFhCDCXKXfDx60fii+syp5iMs3UcNX/1H4zJNgmqSejfhHrYw==' 'sha512-RBYr6Ld4w1yVqaACrgrBLQfPgGhj/1jyacA74WxJ1KM6KVcSWymwrdDwb3HDcdpwiNJ5yssot1He0U9vXoQVlg==' 'sha256-aWZ3y/RxbBYKHXH0z8+8ljrHG1mSBvyzSfxSMjBSaXk=' 'sha256-vOgyKS2vkH4n5TxBJpeh9SgzrE6LVGsAeOAvEST6oCc='; style-src 'self' https://*.netlify.app data: 'unsafe-inline'
  X-Frame-Options: SAMEORIGIN
  Referrer-Policy: strict-origin
  Feature-Policy: geolocation 'self'
  Cache-Control: public, max-age=31536000
  Access-Control-Allow-Origin: *
```

**Reply by FanYueee** [reply]
Now I have embedded the external image link in the docs page, but the following situation occurs

Refused to load the image 'I've hidden this link' because it violates the following Content Security Policy directive: "img-src 'self' https://*.netlify.app data: https://i.giphy.com".

I often add it at the end
; img-src 'self' https://*.netlify.app data: https://i.giphy.com
but it doesn't have any effect

**Reply by h-enk** [reply]
Changing the order should do it: `https://i.giphy.com data:`

See also:
- https://github.com/h-enk/getdoks.org/blob/master/layouts/index.headers#L5
- https://getdoks.org/docs/how-to/images-and-media/images/#add-a-remote-image
